egs2/itako/svs1/local/prep_segments.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+import argparse
+import math
+import os
+import sys
+
+"""Generate segments according to label."""
+
+
+class LabelInfo(object):
+    def __init__(self, start, end, label_id):
+        self.label_id = label_id
+        self.start = start
+        self.end = end
+
+
+class SegInfo(object):
+    def __init__(self):
+        self.segs = []
+        self.start = -1
+        self.end = -1
+
+    def add(self, start, end, label):
+        start = float(start)
+        end = float(end)
+        if self.start < 0 or self.start > start:
+            self.start = start
+        if self.end < end:
+            self.end = end
+        self.segs.append((start, end, label))
+
+    def split(self, threshold=30):
+        seg_num = math.ceil((self.end - self.start) / threshold)
+        if seg_num == 1:
+            return [self.segs]
+        avg = (self.end - self.start) / seg_num
+        return_seg = []
+
+        start_time = self.start
+        cache_seg = []
+        for seg in self.segs:
+            cache_time = seg[1] - start_time
+            if cache_time > avg:
+                return_seg.append(cache_seg)
+                start_time = seg[0]
+                cache_seg = [seg]
+            else:
+                cache_seg.append(seg)
+
+        return_seg.append(cache_seg)
+        return return_seg
+
+
+def pack_zero(file_id, number, length=4):
+    number = str(number)
+    return file_id + "_" + "0" * (length - len(number)) + number
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Prepare segments from HTS-style alignment files",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("scp", type=str, help="scp folder")
+    parser.add_argument(
+        "--threshold",
+        type=int,
+        help="threshold for silence identification.",
+        default=30000,
+    )
+    parser.add_argument(
+        "--silence", action="append", help="silence_phone", default=["pau"]
+    )
+    return parser
+
+
+def make_segment(file_id, labels, threshold=30, sil=["pau", "br", "sil"]):
```

Comment:
It is fine to have the customized design for each recipe and that is the reason that we keep the file in local. But could you possibly add some comments/guidelines/suggestions to others (maybe in the recipe level readme or in the general readme in the template)? Many thanks!

Reply:
This is not an big issue, we can fix it with Yuxun's PR